### ISSUE TITLE
Topology for Kaby Lake Lenovo X1 Carbon 5th

### ISSUE THREAD

**Issue by maxd-nordic** [thesofproject/avs-topology-xml#6]
Hi! Would it be possible to add a configuration for my Thinkpad variant?
It has the PCI device id `0x9d71`. I gave your [guide](https://gist.github.com/crojewsk/4e6382bfb0dbfaaf60513174211f29cb) a try, but my ACPI table doesn't contain an NHLT entry.
[lspci.txt](https://github.com/thesofproject/avs-topology-xml/files/10777107/lspci.txt)
[dmidecode.txt](https://github.com/thesofproject/avs-topology-xml/files/10777111/dmidecode.txt)
[dmesg.txt](https://github.com/thesofproject/avs-topology-xml/files/10777112/dmesg.txt)

**Comment by cujomalainey** [comment]
@crojewsk this is not a Chromebook FYI. Not sure what support you have on your end for this.

**Comment by crojewsk-intel** [comment]
@cujomalainey indeed. Moreover, I'm a bit surprised by the request. The following explains why:

Logs indicate the driver of choice here is `snd_hda_intel` which is neither `snd_soc_skl` nor `snd_soc_avs`. The `snd_hda_intel` probed successfully and a number of endpoints have been enumerated:
```
[    4.078978] input: HDA Intel PCH Mic as /devices/pci0000:00/0000:00:1f.3/sound/card0/input11
[    4.079036] input: HDA Intel PCH Headphone as /devices/pci0000:00/0000:00:1f.3/sound/card0/input12
[    4.079087] input: HDA Intel PCH HDMI/DP,pcm=3 as /devices/pci0000:00/0000:00:1f.3/sound/card0/input13
[    4.079138] input: HDA Intel PCH HDMI/DP,pcm=7 as /devices/pci0000:00/0000:00:1f.3/sound/card0/input14
[    4.079188] input: HDA Intel PCH HDMI/DP,pcm=8 as /devices/pci0000:00/0000:00:1f.3/sound/card0/input15
[    4.079242] input: HDA Intel PCH HDMI/DP,pcm=9 as /devices/pci0000:00/0000:00:1f.3/sound/card0/input16
[    4.079300] input: HDA Intel PCH HDMI/DP,pcm=10 as /devices/pci0000:00/0000:00:1f.3/sound/card0/input17
```

If the intention is to switch to a DSP driver, one needs to modify `dsp_driver` parameter of `snd_intel_dspcfg` module e.g.:
```
options snd_intel_dspcfg dsp_driver=4    #to select avs-driver
options snd_intel_dspcfg dsp_driver=2    #to select skylake-driver
```
In case of skylake-driver, there is a known limitation: HDAudio is supported only with Realtek codecs. Device of yours, @maxd-nordic, is equipped with Conexant device instead.

The avs-driver however, supports any HDAudio codec device, including Conexant codecs by default. avs-driver is present on quite recent kernels but is not enabled by default yet.

To summarize: Could you clarify your ask or your problem, @maxd-nordic?

**Comment by maxd-nordic** [comment]
@crojewsk-intel Thanks for the info! Yes, I'd like to use DSP on my device and was a bit confused by all the different things needed to make this work.
